packages/xpath/src/lib/collections/sort.ts

```diff
@@ -38,17 +45,32 @@ class SeededPseudoRandomNumberGenerator implements PseudoRandomNumberGenerator {
 	}
 }
 
-const isInt = (value: number): value is Int => value % 1 === 0;
-
-export const seededRandomize = <T>(values: readonly T[], seed?: number): T[] => {
+export const seededRandomize = <T>(values: readonly T[], seed?: number | bigint): T[] => {
 	let generator: PseudoRandomNumberGenerator;
 
 	if (seed == null) {
 		generator = new UnseededPseudoRandomNumberGenerator();
-	} else if (!isInt(seed)) {
-		throw 'todo not an int';
 	} else {
-		generator = new SeededPseudoRandomNumberGenerator(seed);
+		let finalSeed: number | bigint;
+		// Per issue #49 (https://github.com/getodk/web-forms/issues/49) this is intended to be "bug-or-feature-compatible"
+		// with JavaRosa's implementation; org.javarosa.core.model.ItemsetBinding.resolveRandomSeed takes the .longValue() of
+		// the double produced by randomSeedPathExpr.eval() — see https://github.com/getodk/javarosa/blob/6ce13527c/src/main/java/org/javarosa/core/model/ItemsetBinding.java#L311:L317 .
+		// That results in a 0L when the double is NaN, which happens (for instance) when there
+		// is a string that does not look like a number (which is a problem in itself, as any non-numeric
+		// looking string will then result in the same seed of 0 — see https://github.com/getodk/javarosa/issues/800).
+		// We'll emulate Java's Double -> Long conversion here (for NaN and some other double values)
+		// so that we produce the same randomization as JR.
+
+		// In Java, a NaN double's .longValue is 0
+		if (Number.isNaN(seed)) finalSeed = 0;
+		// In Java, an Infinity double's .longValue() is 2**63 -1, which is larger than Number.MAX_SAFE_INTEGER, thus we'll need a BigInt.
+		else if (seed === Infinity) finalSeed = 2n ** 63n - 1n;
+		// Analogous with the above conversion, but for -Infinity
+		else if (seed === -Infinity) finalSeed = -(2n ** 63n);
+		// A Java double's .longValue drops the fractional part.
+		else if (typeof seed === 'number' && !Number.isInteger(seed)) finalSeed = Math.trunc(seed);
+		else finalSeed = seed;
+		generator = new SeededPseudoRandomNumberGenerator(finalSeed);
```

Comment:
I think this is where I was focused when I suggested cordoning off the JavaRosa compatibility concern. I also think there's a way to do it that feels both natural and idiomatic: as a subclass of `SeededPseudoRandomNumberGenerator`, with a name that clarifies that responsibility beyond the base Park-Miller algorithm. This logic would then fit neatly in its constructor. I'd be thrilled if we could find a way to translate some of these inline comments into a meaningful JSDoc block too, but I wouldn't push too hard for it. A named class would go a long way toward clarity of where the responsibility boundaries are.

(And perhaps it would be clearer still if that base class was renamed to indicate its Park-Miller-only responsibility?)